lib/teleterm/daemon/daemon.go

```diff
@@ -241,17 +240,21 @@ func (s *Service) CreateGateway(ctx context.Context, params CreateGatewayParams)
 }
 
 type GatewayCreator interface {
-	CreateGateway(context.Context, clusters.CreateGatewayParams) (*gateway.Gateway, error)
+	CreateGateway(context.Context, clusters.CreateGatewayParams) (gateway.Gateway, error)
 }
 
 // createGateway assumes that mu is already held by a public method.
-func (s *Service) createGateway(ctx context.Context, params CreateGatewayParams) (*gateway.Gateway, error) {
+func (s *Service) createGateway(ctx context.Context, params CreateGatewayParams) (gateway.Gateway, error) {
 	targetURI, err := uri.ParseGatewayTargetURI(params.TargetURI)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	cliCommandProvider := clusters.NewDbcmdCLICommandProvider(s.cfg.Storage, dbcmd.SystemExecer{})
+	cliCommandProvider, err := s.cfg.CLICommandProviderManager.Get(uri.New(params.TargetURI))
```

Comment:
I have to say I'm not a big fan of how this provider manager looks. It seems to introduce a lot of boilerplate. Take a look at 7198abc94b – I think we can do this without introducing a new struct.

The unit tests pass with my change, but I think the integration tests are failing due to some of the changes on this branch:

```
go test ./integration -run TestTeleterm
```

```
go test ./integration/proxy -run TestALPNSNIProxyDatabaseAccess/teleterm_gateways_cert_renewal
```